hs_file_types/models/netcdf.py

```diff
@@ -580,6 +580,12 @@ def set_file_type(cls, resource, file_id, user):
                         if k == 'subject':
                             logical_file.metadata.keywords = v
                             logical_file.metadata.save()
+                            # update resource level keywords
+                            resource_keywords = [subject.value.lower() for subject in
+                                                 resource.metadata.subjects.all()]
+                            for kw in logical_file.metadata.keywords:
+                                if kw.lower() not in resource_keywords:
+                                    resource.metadata.create_element('subject', value=kw)
```

Comment:
@hyi I think this is covered in the existing test. See the unit test (https://github.com/hydroshare/hydroshare/blob/4b9e60045586feaae10faeb45c7312ec27681abf/hs_core/tests/api/native/test_core_metadata.py#L1331)

Also this line in the create() of the Subject class: https://github.com/hydroshare/hydroshare/blob/4b9e60045586feaae10faeb45c7312ec27681abf/hs_core/models.py#L1302